lib/dataset/aligned_dict.cpp

```diff
@@ -330,27 +337,56 @@ Dict<Key, Value> Dict<Key, Value>::merge_from(const Dict &other) const {
 }
 
 template <class Key, class Value>
-bool Dict<Key, Value>::item_applies_to(const Key &key,
-                                       const Dimensions &dims) const {
-  const auto &val = m_items.at(key);
+bool AlignedDict<Key, Value>::item_applies_to(const Key &key,
+                                              const Dimensions &dims) const {
+  const auto &val = m_items[key];
   return std::all_of(val.dims().begin(), val.dims().end(),
                      [&dims](const Dim dim) { return dims.contains(dim); });
 }
 
 template <class Key, class Value>
-bool Dict<Key, Value>::is_edges(const Key &key,
-                                const std::optional<Dim> dim) const {
+bool AlignedDict<Key, Value>::is_edges(const Key &key,
+                                       const std::optional<Dim> dim) const {
   const auto &val = this->at(key);
   return core::is_edges(m_sizes, val.dims(),
                         dim.has_value() ? *dim : val.dim());
 }
 
-template class SCIPP_DATASET_EXPORT Dict<Dim, Variable>;
-template class SCIPP_DATASET_EXPORT Dict<std::string, Variable>;
-template SCIPP_DATASET_EXPORT bool equals_nan(const Dict<Dim, Variable> &a,
-                                              const Dict<Dim, Variable> &b);
-template SCIPP_DATASET_EXPORT bool
-equals_nan(const Dict<std::string, Variable> &a,
-           const Dict<std::string, Variable> &b);
+template <class Key, class Value>
+core::Dict<Key, Value> union_(const AlignedDict<Key, Value> &a,
+                              const AlignedDict<Key, Value> &b,
+                              std::string_view opname) {
+  core::Dict<Key, Value> out;
+  out.reserve(out.size() + b.size());
```

Comment:
Reserving too much in most cases, but probably does not matter?

Reply:
It shouldn't matter. If we reserve too little, it will grow geometrically and overallocate anyway.